src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -257,11 +289,37 @@ def create_key(self, arg0, arg1=None, arg2=None, names='i,j,k'):
             a = K(v[0])
             b = K(v[1])
 
-        # QuaternionAlgebra(K, a, b)
         else:
+            # QuaternionAlgebra(K, primes, inv_archimedean)
             K = arg0
-            a = K(arg1)
-            b = K(arg2)
+            if category(K) is not NumberFields():
+                raise ValueError("quaternion algbera must be defined over a number field")
+            if isinstance(arg1, list) and isinstance(arg2, list):
+                if not set(arg2).issubset(set([0,QQ(1/2)])):
+                    raise ValueError("list of local invariants specifying ramification should contain only 0 and 1/2")
+                arg1 = list(set(arg1))
+                if not all([p.is_prime() for p in arg1]):
+                    raise ValueError("quaternion algebra constructor requires a list of primes specifying the ramification")
+                if is_RationalField(K):
+                    if len(arg2) > 1 or (len(arg2) == 1 and is_odd(len(arg1) + 2*arg2[0])):
+                        raise ValueError("quaternion algebra over the rationals must have an even number of ramified places")
+                    D = ZZ.ideal_monoid().prod(arg1).gen()
+                    a, b = hilbert_conductor_inverse(D)
+                    a = QQ(a)
+                    b = QQ(b)
+                else:
+                    if len(arg2) != len(K.real_places()):
+                        raise ValueError("must specify ramification at the real places of %s" % K)
+                    if is_odd(len(arg1) + 2 * sum(arg2)):
+                        raise ValueError("quaternion algebra over the rationals must have an even number of ramified places")
+                    fin_places_pari = [I.pari_prime() for I in arg1]
+                    A = pari(K).alginit([2, [fin_places_pari, [QQ(1/2)] * len(fin_places_pari)], arg2])
```

Comment:
Hi, I suggest using alginit with flag=0 (4th argument), to prevent computation of a maximal order. Indeed, you are throwing away the information anyway, so you may as well not compute it.

Reply:
Thank you,

In the future (when orders for quaternion algebras over number fields are more developed), maybe we could offer an optional flag `compute_maximal_order=False`, and store generators of the order if the user wants to work with a maximal order after generating the algebra.